fe/fe-core/src/main/java/com/starrocks/clone/DiskAndTabletLoadReBalancer.java

```diff
@@ -1398,52 +1525,40 @@ private List<Pair<Long, Set<Long>>> getPartitionTablets(Long dbId, Long tableId,
         return result;
     }
 
-    // NOTICE: call this function as little as possible, cause this will get db lock
-    private boolean isTabletHealthy(Long tabletId, TabletMeta tabletMeta, List<Long> aliveBeIds) {
+    private boolean isTabletUnhealthy(OlapTable olapTable, Long tabletId,
+                                      TabletMeta tabletMeta, List<Long> aliveBeIds) {
         GlobalStateMgr globalStateMgr = GlobalStateMgr.getCurrentState();
-        Database db = globalStateMgr.getDbIncludeRecycleBin(tabletMeta.getDbId());
-        if (db == null) {
-            return false;
+        // Won't hold the db lock, meta change will cause this clone task failed, this is acceptable.
```

Comment:
If remove the db lock here, the following get method should be thread safe, but some of them are not.

Reply:
we can accept the risk here, as the comment said

Reply:
fixed